apps/antalmanac/src/hooks/useQuickSearch.tsx

```diff
@@ -11,13 +13,24 @@ export function useQuickSearch() {
     const navigate = useNavigate();
 
     return useCallback(
-        (deptValue: string, courseNumber: string, termValue: string) => {
+        (deptValue: string, courseNumber: string, termValue: string, sectionCode = '') => {
             const queryParams = {
                 term: termValue,
                 deptValue: deptValue,
                 courseNumber: courseNumber,
             };
 
+            if ((!termValue || !deptValue || !courseNumber) && sectionCode.length > 0) {
+                const course: ScheduleCourse = (AppStore.getAddedCourses() as unknown as ScheduleCourse[]).filter(
```

Comment:
Why is this cast necessary. I don' see any errors here.
```suggestion
                const course: ScheduleCourse = AppStore.getAddedCourses().filter(
```

Reply:
I was following copilots orders to appease the linter